File: testshuffle.py
import random
import copy

def shuffleArray(array):
    shuffled = copy.copy(array)
    for i in range(len(shuffled) - 1, 0, -1):
        j = random.randint(0, i)
        shuffled[i], shuffled[j] = shuffled[j], shuffled[i]
    return shuffled

options = [{'text': 'A'}, {'text': 'B'}, {'text': 'C'}, {'text': 'D'}]

print("Original before:", [o['text'] for o in options])
s1 = shuffleArray(options)
print("Shuffled 1:", [o['text'] for o in s1])
print("Original after 1:", [o['text'] for o in options])
s2 = shuffleArray(options)
print("Shuffled 2:", [o['text'] for o in s2])

File: testshuffle.js
const shuffleArray = (array) => {
    let shuffled = [...array];
    for (let i = shuffled.length - 1; i > 0; i--) {
        const j = Math.floor(Math.random() * (i + 1));
        [shuffled[i], shuffled[j]] = [shuffled[j], shuffled[i]];
    }
    return shuffled;
};

let options = [{ text: "A" }, { text: "B" }, { text: "C" }, { text: "D" }];

console.log("Original before:");
console.log(options.map(o => o.text));

console.log("Shuffled 1:");
console.log(shuffleArray(options).map(o => o.text));

console.log("Original after 1:");
console.log(options.map(o => o.text));

console.log("Shuffled 2:");
console.log(shuffleArray(options).map(o => o.text));
